components/Editor.js

```diff
@@ -96,11 +96,20 @@ class Editor extends React.Component {
 
   carbonNode = React.createRef()
 
-  updateState = updates =>
-    this.setState(updates, () => !this.gist && this.props.onUpdate(this.state))
+  updateState = updates => {
+    this.setState(updates, () => {
+      // eslint-disable-next-line no-unused-vars
+      const { highlights, ...state } = this.state
+      !this.gist && this.props.onUpdate(state)
+    })
+  }
 
   updateCode = code => this.updateState({ code })
 
+  updateTheme = theme => this.updateState({ theme })
```

Comment:
Is it possible to just combine `updateTheme` and `updateHighlights` into a single call since they are always run together? 

As a fallback you could even do `updateTheme(theme, highlights)`

Reply:
I think this is going to simplify the rest of the changes as well, so going to hold off reviewing that until this is updated.